apollo-router/src/plugins/fleet_detector.rs

```diff
@@ -0,0 +1,126 @@
+use schemars::JsonSchema;
+use serde::Deserialize;
+use sysinfo::System;
+use tokio::task::JoinHandle;
+use tower::BoxError;
+use tracing::debug;
+use tracing::info;
+
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+
+#[derive(Debug)]
+struct FleetDetector {
+    handle: JoinHandle<()>,
+}
+
+#[derive(Debug, Default, Deserialize, JsonSchema)]
+struct Conf {}
+
+#[async_trait::async_trait]
+impl Plugin for FleetDetector {
+    type Config = Conf;
+
+    async fn new(_: PluginInit<Self::Config>) -> Result<Self, BoxError> {
+        debug!("beginning environment detection");
+        debug!("spawning continuous detector task");
+        let handle = tokio::task::spawn(async {
+            let mut sys = System::new_all();
+            let mut interval = tokio::time::interval(tokio::time::Duration::from_secs(60));
+            loop {
+                interval.tick().await;
+                sys.refresh_cpu_all();
+                sys.refresh_memory();
+                detect_cpu_values(&sys);
+                detect_memory_values(&sys);
+            }
+        });
+
+        Ok(FleetDetector { handle })
+    }
+}
+
+impl Drop for FleetDetector {
+    fn drop(&mut self) {
+        self.handle.abort();
+    }
+}
+
+fn detect_cpu_values(system: &System) {
+    let cpus = system.cpus();
+    let cpu_count = detect_cpu_count(system);
+    let cpu_freq = cpus.iter().map(|cpu| cpu.frequency()).sum::<u64>() / cpus.len() as u64;
+    info!(value.apollo.router.instance.cpu_freq = cpu_freq);
+    info!(counter.apollo.router.instance.cpu_count = cpu_count);
```

Comment:
We're not using this mechanism for metrics anymore. What you're looking for is `u64_counter!` macro (search in the codebase). And a gauge like [here](https://github.com/apollographql/router/blob/dev/apollo-router/src/query_planner/bridge_query_planner_pool.rs#L45:L45) for the gauge please make sure to always keep a reference somewhere it should never be dropped